inference/main.py

```diff
@@ -132,22 +137,74 @@ def get_servo_states(hal: HAL) -> None:
 
 
 def set_servo_positions(hal: HAL) -> None:
+    """
+    input -> desired positions (radians)
+    output -> set servo positions (degrees)
+    """
     positions_deg = []
     for joint in joints:
-        # convert from radians to degrees
+        # Convert from radians to degrees and apply offset
         desired_pos_deg = math.degrees(joint.desired_position)
         desired_pos_deg += joint.offset_deg
         positions_deg.append((joint.servo_id, desired_pos_deg))
 
-    # print(f"[INFO]: SET servo positions (deg): {positions_deg}")
-
     if MOCK:
         return
 
     hal.servo.set_positions(positions_deg)
 
 
-def inference(policy: ort.InferenceSession, hal: HAL, cfg: Sim2simCfg, data_queue: mp.Queue) -> None:
+def init_madgwick():
+    # Initialize Madgwick filter with increased beta for accelerometer weight
+    return Madgwick(frequency=100.0, beta=0.7)
+
+
+def imu_data_to_numpy(imu_data):
+    """
+    Converts IMU data to numpy arrays for gyro and accelerometer.
+    """
+    gyro = imu_data['gyro']
+    accel = imu_data['accel']
+    # Convert to proper units: gyro to rad/s, accel to g
+    gyro_array = np.array([gyro['x'], gyro['y'], gyro['z']]) * np.pi / 180.0  # deg/s to rad/s
+    accel_array = np.array([accel['x'], accel['y'], accel['z']]) / 1000.0     # mg to g
+    return gyro_array, accel_array
+
+
+class IMUHandler:
```

Comment:
This should be taken outside the inference logic and later on moved to 'backend' (https://github.com/Zeroth-Robotics/OpenLCH/issues/41)